test/e2e/tests/test_hpo.py

```diff
@@ -18,40 +18,55 @@
 import logging
 from typing import Dict
 
-from e2e import service_marker, create_sagemaker_resource
-from e2e.replacement_values import REPLACEMENT_VALUES
-from acktest.resources import random_suffix_name
+from acktest.resources import random_suffix_name, load_resource_file
 from acktest.k8s import resource as k8s
+from e2e import (
+    resource_directory,
+    CRD_GROUP,
+    CRD_VERSION,
+    service_marker,
+    create_sagemaker_resource,
+    wait_for_status,
+)
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e.bootstrap_resources import get_bootstrap_resources
+from time import sleep
 
 RESOURCE_PLURAL = "hyperparametertuningjobs"
-HPO_JOB_STATUS_CREATED = ("InProgress", "Completed")
-HPO_JOB_STATUS_STOPPED = ("Stopped", "Stopping")
 
+def _sagemaker_client():
+    return boto3.client('sagemaker')
 
-@pytest.fixture(scope="module")
-def xgboost_hpojob():
+def _make_hpojob():
     resource_name = random_suffix_name("xgboost-hpojob", 32)
 
     replacements = REPLACEMENT_VALUES.copy()
     replacements["HPO_JOB_NAME"] = resource_name
 
-    reference, spec, resource = create_sagemaker_resource(
-        resource_plural=RESOURCE_PLURAL,
-        resource_name=resource_name,
-        spec_file="xgboost_hpojob",
-        replacements=replacements,
+    data = load_resource_file(
+        resource_directory, "xgboost_hpojob", additional_replacements=replacements
+    )
+
+    reference = k8s.CustomResourceReference(
+        CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL, resource_name, namespace="default"
     )
-    assert resource is not None
-    yield (reference, resource)
 
-    # Delete the k8s resource if not already deleted by tests
-    if k8s.get_resource_exists(reference):
-        k8s.delete_custom_resource(reference)
+    return reference, data
+
+@pytest.fixture(scope="function")
+def xgboost_hpojob():
+    hpo_job, data = _make_hpojob()
+    resource = k8s.create_custom_resource(hpo_job, data)
```

Comment:
in all tests,

1. use `create_sagemaker_resource` to reduce amount of code. basically, this test doesnt need make_hpojob at all

2. and instead of 
  ```
  if k8s.get_resource_exists(hpo_job):
          k8s.delete_custom_resource(hpo_job)
  ```
  do
  ```
  if k8s.get_resource_exists(reference):
      _, deleted = k8s.delete_custom_resource(reference)
      assert deleted
  ```

Reply:
Yea, this was one of the changes i said I saw after a more recent rebase. Done.